include/benchmark/benchmark_api.h

```diff
@@ -493,8 +525,40 @@ class State {
         internal::ThreadManager* manager);
 
  private:
-  void StartKeepRunning();
-  void FinishKeepRunning();
+  friend class internal::StateIterator;
+
+  size_t GetBatch(size_t batch_size, bool atomic_batch) {
```

Comment:
boolean parameters can cause callsite readability issues. it's usually better to define an enum instead:

```
enum class AtomicBatch { kNo, kYes };

int GetBatch(int size, AtomicBatch atomic);

...

GetBatch(n, AtomicBatch::kYes);
```